src/traces/bar/plot.js

```diff
@@ -530,5 +519,32 @@ function calcTextinfo(calcTrace, index, xa, ya) {
         if(hasFlag('final')) text.push(formatNumber(final));
     }
 
+    if(trace.type === 'funnel') {
+        if(hasFlag('value')) text.push(formatNumber(cdi.s));
+
+        var nPercent = 0;
+        if(hasFlag('percent initial')) nPercent++;
+        if(hasFlag('percent previous')) nPercent++;
+        if(hasFlag('percent total')) nPercent++;
+
+        var hasMultiplePercents = nPercent > 1;
+
+        if(hasFlag('percent initial')) {
+            tx = formatNumber(Math.round(100 * cdi.begR), true) + '%';
+            if(hasMultiplePercents) tx += ' of initial';
+            text.push(tx);
+        }
+        if(hasFlag('percent previous')) {
+            tx = formatNumber(Math.round(100 * cdi.difR), true) + '%';
+            if(hasMultiplePercents) tx += ' of previous';
+            text.push(tx);
+        }
+        if(hasFlag('percent total')) {
+            tx = formatNumber(Math.round(100 * cdi.sumR), true) + '%';
```

Comment:
Why are you using the `hover:true` option here?

Reply:
Otherwise the correct text didn't show up possibly related to the hidden axis?

Reply:
> Otherwise the correct text didn't show up possibly related to the hidden axis?

Probably because we don't coerce `tickformat`, `ticksuffix` and `tickprefix` when `ax.visible: false`. We should fix that.

Reply:
Actually this was not related to hidden axes. We need to pass `true` to `ticktext` function to enable high precision and similarly for `waterfalls` as done in  https://github.com/plotly/plotly.js/pull/3817/commits/cad2728b964a373cd7593f93217501c4588b1799.